ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Offset.scala

```diff
@@ -29,6 +29,37 @@ final case class Offset(bytes: Bytes) extends Ordered[Offset] {
   def toInputStream: InputStream = bytes.toInputStream
 
   def toHexString: Ref.HexString = bytes.toHexString
+
+  /** Returns the next smaller offset (lexicographically).
+    *
+    * NOTE: The usage of this method is correct under the assumption that the string length of any offset
```

Comment:
@meiersi-da @mziolekda This is a constraint I did not take in consideration when creating this task. After offline discussions, the code/reasoning simplification and performance it brings is negligible. Hence, this PR will be dropped. Please do let me know if you have a better idea.